frontend/controller/actions/identity.js

```diff
@@ -369,24 +360,128 @@ export default (sbp('sbp/selectors/register', {
       })()
       : []
 
+    // Used to store stateEncryptionKeyId, salt, encryptedStateEncryptionKey
+    const extraLoginAttributes = {}
+
     try {
-      sbp('appLogs/startCapture', username)
-      const state = await sbp('gi.db/settings/load', username)
-      let contractIDs = []
+      // Utility class to represent an empty state during error handling
+      class EmptyState extends Error {}
```

Comment:
Creating classes in the middle of a function is not clean code.

If you must create a new class for the error, please put it in the appropriate place along with the other `GIError*` classes